firefox-ios/firefox-ios-tests/Tests/ClientTests/SummarizationCheckerTests.swift

```diff
@@ -0,0 +1,90 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+@testable import Client
+import XCTest
+
+final class SummarizationCheckerTests: XCTestCase {
+  func testSummarizationCheckerCanSummarizeTrue() throws {
+    let jsStub: [String: Any] = [
+      "canSummarize": true,
+      "wordCount": 42,
+    ]
+
+    let result = try SummarizationChecker.parse(jsStub)
+    XCTAssertTrue(result.canSummarize)
+    XCTAssertNil(result.reason)
+    XCTAssertEqual(result.wordCount, 42)
+  }
+
+  /// JS stub for a page that is not reader‑readable.
+  func testSummarizationCheckerCanSummarizeFalseDocumentNotReady() throws {
+    let jsStub: [String: Any] = [
+      "canSummarize": false,
+      "reason": "documentNotReady",
+      "wordCount": 0,
+    ]
```

Comment:
Do you maybe wanna make it a jsStub builder maybe

Reply:
Done in https://github.com/mozilla-mobile/firefox-ios/pull/28022/commits/9e97646fa7a9dba3017ce92d043fd684fb11b5b3